internal/consul/lock.go

```diff
@@ -0,0 +1,70 @@
+package consul
+
+import (
+	"fmt"
+	"time"
+
+	consulApi "github.com/hashicorp/consul/api"
+)
+
+const (
+	sessionName      string = "trento-session"
+	monitorRetries   int    = 10
+	monitorRetryTime        = 100 * time.Millisecond
+	waitTime                = 15 * time.Second
+)
+
+// Created to enable unit testing
+var lockGet = func(c *client, prefix string, q *consulApi.QueryOptions) (*consulApi.KVPair, *consulApi.QueryMeta, error) {
```

Comment:
(nit) functions / methods should have the verb first:
```suggestion
var getLock = func(c *client, prefix string, q *consulApi.QueryOptions) (*consulApi.KVPair, *consulApi.QueryMeta, error) {
```